app/src/main/java/org/oppia/app/topic/overview/TopicOverviewViewModel.kt

```diff
@@ -20,9 +20,22 @@ class TopicOverviewViewModel @Inject constructor() : ObservableViewModel() {
 
   var downloadStatusIndicatorDrawableResourceId = ObservableField(R.drawable.ic_available_offline_primary_24dp)
 
-  /** Returns the number of megabytes of disk space this topic requires, formatted for display. */
-  fun getTopicSizeMb(): String {
-    val topicSizeMb: Double = (topic.get()?.diskSizeBytes ?: 0) / (1024.0 * 1024.0)
-    return decimalFormat.format(topicSizeMb)
+  /** Returns the space this topic requires on disk, formatted for display. */
+  fun getTopicSizeWithUnit(): String {
+    val size: Double = topic.get()?.diskSizeBytes!!.toDouble()
+    if (size == 0.0) {
+      return "0 KB"
+    }
+    val sizeInKB = size / 1024.0
+    val sizeInMB = size / 1024.0 / 1024.0
+    val sizeInGB = size / 1024.0 / 1024.0 / 1024.0
+    val sizeInTB = size / 1024.0 / 1024.0 / 1024.0 / 1024.0
+    return when {
+      sizeInTB >= 1 -> decimalFormat.format(sizeInTB) + " TB"
```

Comment:
Let's default to KB for now since otherwise we need strings for each of these for localization.

Reply:
okay, then KB will be the only unit for now.

Reply:
As per the conversation, we are keeping this, and we should keep these and shift the units to `strings.xml`